### PR TITLE
Use AsDriver and DistinctUntilChanged in SignUpViewModel

### PR DESCRIPTION
**Pull request by arturs105**
## What's this?
Does what the title says and also adds shake behaviour based on a subject because `DistinctUntilChanged` calls make it no possible to rely on error messages to shake the email field.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Core.UI/ViewModels/SignUpViewModel.cs [hunk 1]
```diff
@@ -196,6 +199,8 @@ private void onError(Exception exception)
             if (errorHandlingService.TryHandleDeprecationError(exception))
                 return;
 
+            shakeEmailFieldSubject.OnNext(Unit.Default);
```
Comment (sofokli1) [inline Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: shouldn't this be in the `EmailIsAlreadyUsedException` case?

File: Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs [hunk 2]
```diff
@@ -162,6 +162,55 @@ public void EmitsTheOppositeValueWhenTogglePasswordVisibilityActionIsExecuted(in
...
+                    .Returns(
+                        Observable.Throw<Unit>(exception)
+                    );
+                var observer = TestScheduler.CreateObserver<Unit>();
+                ViewModel.ShakeEmailField.Subscribe(observer);
+
+                ViewModel.SignUp.Execute();
+                TestScheduler.Start();
+
+                observer.Values().Should().HaveCount(1);
```
Comment (sofokli1) [inline Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs]: can you add a test where you check that you do not get a shake. Like correct signup or other kind of exception that should not trigger a shake.

File: Toggl.Core.UI/ViewModels/SignUpViewModel.cs [hunk 3]
```diff
@@ -143,7 +142,11 @@ private async Task signUp()
             else
                 passwordErrorSubject.OnNext(string.Empty);
 
-            if (!credentialsAreValid)  return;
+            if (!credentialsAreValid)
+            {
+                shakeEmailFieldSubject.OnNext(Unit.Default);
```
Comment (sofokli1) [inline Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: Won't this shake the email if the password is the wrong size? Isn't that a bit confusing?

Comment (arturs105) [reply Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: Me and Janika discussed this a while ago (when i was writing the `SignUpViewModel`) and decided to shake just the email field for any error message, be it email error, password error, or sign in error.

File: Toggl.Core.UI/ViewModels/SignUpViewModel.cs (hunk 1 above)
Comment (arturs105) [reply Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: Same explanation as below - the email field is shaken for any displayed error message.

File: Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs [hunk 4]
```diff
@@ -209,6 +209,71 @@ public void EmitsWhenTryingToSignUpWithAlreadySignedUpEmail()
...
+                ErrorHandlingService.TryHandleDeprecationError(exception).Returns(true);
+                var observer = TestScheduler.CreateObserver<Unit>();
+                ViewModel.ShakeEmailField.Subscribe(observer);
+
+                ViewModel.SignUp.Execute();
+                TestScheduler.Start();
+
+                observer.Values().Should().BeEmpty();
+
+
```
Comment (sofokli1) [inline Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs]: 🔥, up to you

**Pull request merged**